hyperspy/axes.py

```diff
@@ -112,11 +112,38 @@ def _convert_compact_units(self, factor=0.25, inplace=True):
 
     _convert_compact_units.__doc__ %= FACTOR_DOCSTRING
 
+    def _get_value_from_value_with_units(self, value):
+        "Get axes value by passing a calibrated value with a unit, like '20nm'"
+        if isinstance(value, (np.ndarray, list, tuple)):
+            value = np.asarray([_ureg.parse_expression(val).to(self.units).magnitude for val in value])
+        else:
+            value = _ureg.parse_expression(value)
+            if not hasattr(value, 'units'):
+                raise ValueError('"{}" should contain a unit.'.format(value))
+            value = value.to(self.units).magnitude
+        return value
+
     def _get_index_from_value_with_units(self, value):
-        value = _ureg.parse_expression(value)
-        if not hasattr(value, 'units'):
-            raise ValueError('"{}" should contains an units.'.format(value))
-        return self.value2index(value.to(self.units).magnitude)
+        "Get axes index by passing a calibrated value with a unit, like '20nm'"
+        return self.value2index(self._get_value_from_value_with_units(value))
+
+    def _get_value_from_relative_string(self, relative_string):
+        "Get axes index by relative indexing using string like 'rel0.5'"
+        try:
+            assert relative_string.startswith('rel')
+            relative_value = float(relative_string[3:])
+        except (ValueError, AssertionError):
+           print(f"Expected relative units of kind 'rel0.2' but '{relative_string}' could not be interpreted this way.")
+           raise
```

Comment:
I am not sure this is really necessary, since this is a private method and it is called after `startswith('rel')` has been checked. To keep the code more readable, I would remove it. 
However, it would be good to check that what's follow `rel` is a number between 0 and 1.